yb-voyager/cmd/importData.go

```diff
@@ -798,6 +801,60 @@ func importData(importFileTasks []*ImportFileTask, errorPolicy importdata.ErrorP
 	fmt.Printf("\nImport data complete.\n")
 }
 
+// For a fresh start but non empty tables in tableList && OnPrimaryKeyConflict is set to IGNORE -> notify user
+func checkPKConflictModeOnFreshStart(importFileTasks, pendingTasks, completedTasks []*ImportFileTask) {
+	if !isFreshStart(importFileTasks, pendingTasks, completedTasks) {
+		log.Infof("Not a fresh start, skipping primary key conflict mode check.")
+		return
+	}
+
+	// in case of ERROR mode, no need to check for non-empty tables
+	// but for IGNORE or UPDATE(in future), we need to prompt user
+	if tconf.OnPrimaryKeyConflictAction == constants.PRIMARY_KEY_CONFLICT_ACTION_ERROR {
+		return
+	}
+
+	pendingTablesList := importFileTasksToTableNameTuples(pendingTasks)
+	tablesHavingData := tdb.GetNonEmptyTables(pendingTablesList)
+	if len(tablesHavingData) == 0 {
+		return
+	}
+
+	allNonPKTables := true
+	for _, table := range tablesHavingData {
+		colList, err := tdb.GetPrimaryKeyColumns(table)
+		if err != nil {
+			utils.ErrExit("Failed to get primary key columns for table %s: %s", table.ForOutput(), err)
+		}
+		if len(colList) != 0 {
+			allNonPKTables = false
+			log.Infof("found table with PK: %s, no need to prompt user", table.ForOutput())
+			continue
+		}
+	}
+	if allNonPKTables {
+		return // no PK tables, so no need to prompt user
+	}
+
+	utils.PrintAndLog(
+		"Target tables with pre-existing data: %v\n\n"+
+			"Selected mode: IGNORE on primary-key conflicts\n"+
+			"This mode offers improved import speed when your data contains no PK constraint violations.\n\n"+
+			"Under this mode:\n"+
+			"  • Duplicate Primary Key rows are silently skipped.\n"+
+			"  • Import proceeds without error for skipped entries.\n"+
+			"  • Only unique rows will be imported.\n\n", sqlname.NameTupleListToStrings(tablesHavingData),
```

Comment:
I'm wondering: do we need to exclude the non-pk tables from this list as well? since it's basically not applicable?

Reply:
yes good catch